lib/components/viewers/route-viewer.js

```diff
@@ -51,55 +79,121 @@ class RouteViewer extends Component {
       transitOperators,
       viewedRoute
     } = this.props
-    const sortedRoutes = routes
-      ? Object.values(routes).sort(
-        coreUtils.route.makeRouteComparator(transitOperators)
+    const { agency, mode, search } = this.state
+    const routesArray = routes ? Object.values(routes) : []
+
+    const filteredAndSortedRoutes = routesArray
+      .filter(
+        (route) =>
+          // If the filter isn't defined, don't check.
+          (!agency || agency === route.agencyName) &&
+          (!mode || mode === route.mode) &&
+          // If user search is active, filter here
+          (!search ||
+            ((route.longName && route.longName.includes(search)) ||
+              (route.shortName && route.shortName.includes(search))))
+      )
+      .sort(coreUtils.route.makeRouteComparator(transitOperators))
+
+    // get agency and mode lists from route list
+    const agencies = Array.from(
+      new Set(routesArray.map((route) => route.agencyName || route.agency.name))
+    )
+      .filter((agency) => agency !== undefined)
+      .sort()
+
+    const modes = Array.from(
```

Comment:
Can this `modes` variable filter the available options based on only the routes for the presently selected agency? E.g., if Cobb is selected, only show bus routes (rather than bus, subway, and tram).

Reply:
Handled in e886cfc75dde3dbec1844daa3cd88c51a0750e23